languages/js/src/Host.ts

```diff
@@ -343,7 +410,13 @@ export class Host {
         );
         return { value: { Dictionary: { fields } } };
       default:
-        const instance_id = this.cacheInstance(v);
+        // TODO (dhatch): Reliable check that this is a class?
+        let instanceId = undefined;
+        if (v instanceof Function) {
```

Comment:
This clearly works but curious why the check is for `Function`?

Reply:
A class is a function in Javascript. There is no class type in the actual language, only one in TypeScript (that we've supplied).